theano/tensor/tests/test_opt.py

```diff
@@ -3556,6 +3557,7 @@ def test_equality_shapes(self):
             f = theano.function([x], T.eq(g, 0))
             assert f([3, 3]) == 0
             assert f([]) == 1
+            self.assertTrue(check_stack_trace(f, ops_to_check='last'))
```

Comment:
Wait, which optimization is supposed to be tested here? What does the final graph look like?